android/src/main/kotlin/com/sap/gigya_flutter_plugin/GigyaFlutterPlugin.kt

```diff
@@ -13,15 +13,23 @@ import io.flutter.plugin.common.MethodChannel.Result
 /** GigyaFlutterPlugin */
 class GigyaFlutterPlugin : FlutterPlugin, MethodCallHandler {
 
-    private lateinit var sdk: GigyaSDKWrapper<*>
+    companion object {
+        private lateinit var sdk: GigyaSDKWrapper<*>
+        fun <T : GigyaAccount> init(application: Application, accountObj: Class<T>) {
+            if (::sdk.isInitialized.not()) {
+                sdk = GigyaSDKWrapper(application, accountObj)
+            }
+        }
+    }
+
 
     /// Main communication method channel.
     private lateinit var channel: MethodChannel
 
     private var _messenger: BinaryMessenger? = null
 
     override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
-        sdk = GigyaSDKWrapper(flutterPluginBinding.applicationContext as Application, GigyaAccount::class.java)
+        init(flutterPluginBinding.applicationContext as Application, GigyaAccount::class.java)
```

Comment:
This is a good solution. 
I will need to do some testing on our core SDKs just to make sure everything is in sync.
Looks good.

Reply:
This method is problematic here because of the current core SDK init.
Once you initialize a custom scheme you cannot override it with a different one. Therefore you can only call init with a custom schema once (it will give an error if you call it twice with a different schema).
I am checking the option to update the core SDK to remove this restriction and then this update will be relevant to the new core SDK.

Sorry for these delays. Trying to resolve this as soon as possible.

Reply:
@tal-mi I do not see a problem. this initialization is performed once when the application is created and each time after the system kills it...

Reply:
@tal-mi maybe you miss this check:
```
if (::sdk.isInitialized.not()) {                
   sdk = GigyaSDKWrapper(application, accountObj)
}
```

Reply:
The first init will be called in the application class if the client uses it.
The second one will be called when "onAttachedToEngine" is fired with "GigyaAccount::class.java" explicitly.
The core SDK will not allow reinitialization with different schema classes (it will produce an error if the schema classes differ).

Reply:
Yup I missed it...